src/Database/Driver/Mysql.php

```diff
@@ -223,6 +251,39 @@ public function supportsDynamicConstraints(): bool
         return true;
     }
 
+    /**
+     * Returns true if the connected server is MariaDB.
+     *
+     * @return bool
+     */
+    public function isMariadb(): bool
+    {
+        $this->version();
+
+        return $this->serverType === 'mariadb';
+    }
+
+    /**
+     * Returns connected server version.
+     *
+     * @return string
+     */
+    public function version(): string
+    {
+        if ($this->_version === null) {
+            $this->connect();
+            $this->_version = (string)$this->_connection->getAttribute(PDO::ATTR_SERVER_VERSION);
+
+            if (strpos($this->_version, 'MariaDB') !== false) {
+                $this->serverType = 'mariadb';
+                $parts = explode(':', $this->_version);
+                $this->_version = array_pop($parts);
```

Comment:
For what kind of version string format is this intended?

From my experience the MariaDB version string usually looks something like `5.5.5-10.3.23-MariaDB` sometimes with additional trailing information like `-0ubuntu0.16.04.1`.

Reply:
Testing on Debian 9 I got string like `5.5.5-10.3.23-MariaDB-1:10.3.23+maria~stretch`.

I guess we can update the check to account for `5.5.5-10.3.23-MariaDB` too.

Reply:
@ndm2 I added a `debug()` to show the version and on the Ubuntu 18.04 VM also it shows similar format `5.5.5-10.4.13-MariaDB-1:10.4.13+maria~focal`.

Reply:
Sorry the server used isn't one provided through `apt` on Ubuntu 18.04 but a dockerized on provided by the `getong/mariadb-action` action, though the point remains the same.

Reply:
On Debian 9 

|`$pdo->getAttribute(PDO::ATTR_SERVER_VERSION)` |`SELECT VERSION()`                       |
|-----------------------------------------------|-----------------------------------------|
|`5.5.5-10.3.23-MariaDB-1:10.3.23+maria~stretch`|`10.3.23-MariaDB-1:10.3.23+maria~stretch`|

Reply:
I can check it again later, but from my testing when I implemented the CTE stuff, my local Ubuntu VM produced the `...-MariaDB-0ubuntu0.16.04.1` variant, and on Windows it was just `...-MariaDB`.

Reply:
I have updated the code to use regex match to account for both possible formats.

Reply:
`5.5.5-` is hardcoded and always returned by MariaDB

https://github.com/MariaDB/server/commit/c50ee6c23dbeb090963580754bec2f0a96ac0557

https://github.com/php/php-src/commit/f9f4a68368406a2ba2bd9fab0494261090183e8e

Reply:
@garas Yup. The regex used should account for all known possible strings format mentioned in the comments above.